Typescript: Type 'InternalDiscordGatewayAdapterCreator' is not assignable to type 'DiscordGatewayAdapterCreator'.

### Which package is this bug report for?

voice

### Issue description


Using field "voiceAdapterCreator" from Guild when calling joinVoiceChannel results in a typescript compilation error.
```error TS2322: Type 'InternalDiscordGatewayAdapterCreator' is not assignable to type 'DiscordGatewayAdapterCreator'.
  Types of parameters 'methods' and 'methods' are incompatible.
    Type 'DiscordGatewayAdapterLibraryMethods' is not assignable to type 'InternalDiscordGatewayAdapterLibraryMethods'.
      Types of property 'onVoiceStateUpdate' are incompatible.
        Type '(data: import("D:/Projects/debil-bot/node_modules/discord-api-types/payloads/v10/voice").GatewayVoiceState) => void' is not assignable to type '(data: import("D:/Projects/debil-bot/node_modules/@discordjs/builders/node_modules/discord-api-types/payloads/v9/voice").GatewayVoiceState) => void'.
          Types of parameters 'data' and 'data' are incompatible.
            Type 'import("D:/Projects/debil-bot/node_modules/@discordjs/builders/node_modules/discord-api-types/payloads/v9/voice").GatewayVoiceState' is not assignable to type 'import("D:/Projects/debil-bot/node_modules/discord-api-types/payloads/v10/voice").GatewayVoiceState'.
              Types of property 'channel_id' are incompatible.
                Type 'string | null' is not assignable to type 'string | undefined'.
                  Type 'null' is not assignable to type 'string | undefined'.
```

Full npm list output:
```$ npm list
├── @discordjs/opus@0.7.0
├── @discordjs/voice@0.9.0
├── @sentry/node@6.19.7
├── @sentry/tracing@6.19.7
├── @types/jest@27.5.0
├── @typescript-eslint/eslint-plugin@5.22.0
├── @typescript-eslint/parser@5.22.0
├── discord.js@13.6.0
├── eslint-config-prettier@8.5.0
├── eslint@8.15.0
├── jest@27.5.1
├── linq@3.2.4
├── play-dl@1.9.4
├── prettier@2.6.2
├── ts-jest@27.1.4
├── ts-node@10.7.0
├── tweetnacl@1.0.3
├── typescript@4.6.4
└── winston@3.7.2
```

### Code sample

```typescript
const channel: VoiceBasedChannel = getChannel(); // set channel
const voiceConnection = joinVoiceChannel({
  guildId: channel.guildId,
  channelId: channel.id,
  adapterCreator: channel.guild.voiceAdapterCreator, //Error here. Guild.voiceAdapterCreator is of type "InternalDiscordGatewayAdapterCreator"
});
```


### Package version

@discordjs/voice@0.9.0

### Node.js version

v16.14.1

### Operating system

Windows 10

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds, GuildVoiceStates, GuildMessages

### I have tested this issue on a development release

_No response_

--- a/packages/builders/package.json
+++ b/packages/builders/package.json
@@ -54,7 +54,7 @@
 	"dependencies": {
 		"@sapphire/shapeshift": "^3.0.0",
 		"@sindresorhus/is": "^4.6.0",
-		"discord-api-types": "^0.31.1",
+		"discord-api-types": "^0.32.1",
 		"fast-deep-equal": "^3.1.3",
 		"ts-mixer": "^6.0.1",
 		"tslib": "^2.3.1"

--- a/packages/discord.js/package.json
+++ b/packages/discord.js/package.json
@@ -52,7 +52,7 @@
     "@discordjs/rest": "workspace:^",
     "@sapphire/snowflake": "^3.2.1",
     "@types/ws": "^8.5.3",
-    "discord-api-types": "^0.31.1",
+    "discord-api-types": "^0.32.1",
     "fast-deep-equal": "^3.1.3",
     "lodash.snakecase": "^4.1.1",
     "tslib": "^2.3.1",

--- a/packages/rest/package.json
+++ b/packages/rest/package.json
@@ -53,7 +53,7 @@
 		"@discordjs/collection": "workspace:^",
 		"@sapphire/async-queue": "^1.3.1",
 		"@sapphire/snowflake": "^3.2.1",
-		"discord-api-types": "^0.29.0",
+		"discord-api-types": "^0.32.1",
 		"tslib": "^2.3.1",
 		"undici": "^5.2.0"
 	},

--- a/packages/voice/package.json
+++ b/packages/voice/package.json
@@ -51,7 +51,7 @@
 	"homepage": "https://discord.js.org",
 	"dependencies": {
 		"@types/ws": "^8.5.3",
-		"discord-api-types": "^0.29.0",
+		"discord-api-types": "^0.32.1",
 		"prism-media": "^1.3.2",
 		"tiny-typed-emitter": "^2.1.0",
 		"tslib": "^2.3.1",

